app/services/users/checkins/import_checkins.rb

```diff
@@ -0,0 +1,43 @@
+module Users::Checkins
+  class ImportCheckins
+    def initialize(params)
+      @file = params[:file]
+      @device = Device.find(params[:device_id])
+    end
+
+    def success?
+      return false unless @file && valid_file?
+      @device.csv = File.open(@file.path, 'r')
+      @device.save
+      import
+      true
+    end
+
+    def valid_file?
+      CSV.foreach(@file.path, headers: true) do |csv|
+        return csv.headers == Checkin.column_names
+      end
+    end
+
+    def import
+      resource_info = Cloudinary::Api.resource(@device.csv.public_id, resource_type: 'raw')
+      csv_file = Cloudinary::Downloader.download(resource_info['secure_url'])
```

Comment:
So here we upload file as device attachment and then download it back before starting the job?